pepper-apis/src/main/java/org/broadinstitute/ddp/route/CheckInvitationRoute.java

```diff
@@ -0,0 +1,71 @@
+import org.broadinstitute.ddp.db.dto.InvitationDto;
+import org.broadinstitute.ddp.db.dto.StudyDto;
+import org.broadinstitute.ddp.json.invitation.CheckInvitationPayload;
+import org.broadinstitute.ddp.util.ValidatedJsonInputRoute;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+public class CheckInvitationRoute extends ValidatedJsonInputRoute<CheckInvitationPayload> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CheckInvitationRoute.class);
+
+    @Override
+    public Object handle(Request request, Response response, CheckInvitationPayload payload) throws Exception {
+        String studyGuid = request.params(RouteConstants.PathParam.STUDY_GUID);
+        String invitationGuid = payload.getInvitationGuid();
+        LOG.info("Attempting to check invitation {} in study {}", invitationGuid, studyGuid);
+
+        int status = TransactionWrapper.withTxn(handle -> {
+            StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyGuid);
+            if (studyDto == null) {
+                LOG.error("Invitation check called for non-existent study with guid {}", studyGuid);
+                return HttpStatus.SC_BAD_REQUEST;
+            }
+
+            List<String> permittedStudies = handle.attach(JdbiClientUmbrellaStudy.class)
+                    .findPermittedStudyGuidsByAuth0ClientIdAndAuth0Domain(payload.getAuth0ClientId(), payload.getAuth0Domain());
+            if (permittedStudies.contains(studyGuid)) {
+                LOG.info("Invitation check by client clientId={}, tenant={}",
+                        payload.getAuth0ClientId(), payload.getAuth0Domain());
+            } else {
+                LOG.error("Invitation check by client which does not have access to study: clientId={}, tenant={}",
+                        payload.getAuth0ClientId(), payload.getAuth0Domain());
+                return HttpStatus.SC_BAD_REQUEST;
+            }
+
+            // todo: check recaptcha
+
+            InvitationDao invitationDao = handle.attach(InvitationDao.class);
+            InvitationDto invitation = invitationDao.findByInvitationGuid(studyDto.getId(), invitationGuid).orElse(null);
+            if (invitation == null) {
+                // It might just be a typo, so do a warn instead of error log.
+                LOG.warn("Invitation {} does not exist", invitationGuid);
+                return HttpStatus.SC_BAD_REQUEST;
+            } else if (invitation.isVoid()) {
+                LOG.error("Invitation {} is voided", invitationGuid);
+                return HttpStatus.SC_BAD_REQUEST;
+            } else if (invitation.isAccepted()) {
+                LOG.error("Invitation {} has already been accepted", invitationGuid);
+                return HttpStatus.SC_BAD_REQUEST;
+            } else {
+                LOG.info("Invitation {} is valid", invitationGuid);
+                return HttpStatus.SC_OK;
```

Comment:
Intent here is to *not* divulge detailed status, but just have a UI say something like "We're sorry, that invitation is invalid" more generally, right?  Looks good.